mathesar_ui/src/component-library/cancel-or-proceed-button-pair/CancelOrProceedButtonPair.svelte

```diff
@@ -34,42 +34,34 @@
   export let onProceed: () => Promise<void>;
   export let canProceed = true;
   export let canCancel = true;
+  export let isProcessing = false;
+  /**
+   * Bind to this function if you want to be able to programmatically call the
+   * proceed function from within the parent component and show the loading
+   * spinner while the promise is resolving.
+   */
```

Comment:
As a pattern, I think it's best if we do not encourage binding to functions we'd like to call from a parent component. Instead, it would be better to bind to the component and call the function using the component reference.

Encouraging binding to a function implies that we expect users to pass their own functions down which is not what we want in this scenario.

Reply:
> Encouraging binding to a function implies that we expect users to pass their own functions down which is not what we want in this scenario.

Right. Good point. Thanks for this feedback. I'll fix that.

Reply:
I opened #1009 to address this